--- a/feeds/morse/essentials/morse_driver/Makefile
+++ b/feeds/morse/essentials/morse_driver/Makefile
@@ -81,11 +81,13 @@
   MORSE_MAKEDEFS += CONFIG_DISABLE_MORSE_RC=y
 endif
 
+MAC80211_VERSION := $(shell sed -n 's/^PKG_VERSION:=\([^ -]*\).*/\1/p' $(TOPDIR)/package/kernel/mac80211/Makefile)
+
 MORSE_MAKEDEFS += \
   MORSE_VERSION=0-$(PKG_VERSION) \
   KERNEL_SRC=$(LINUX_DIR) \
   CONFIG_MORSE_SDIO_ALIGNMENT=$(CONFIG_MORSE_SDIO_ALIGNMENT) \
-  CONFIG_BACKPORT_VERSION=v6.1.24
+  CONFIG_BACKPORT_VERSION=v$(MAC80211_VERSION)
 # This refers to the version of mac80211 backported to OpenWrt
 # Occasionally patches are required to remove some parts of the driver
 # as OpenWrt may sometimes pull in further patches from later kernel versions
