--- a/dot11ah/ie.c
+++ b/dot11ah/ie.c
@@ -526,7 +526,9 @@
 	if (!ies_mask)
 		return 0;
 
-	if (ieee80211_is_s1g_short_beacon(frame_control) ||
+	/* Short S1G beacons omit the beacon compatibility element. */
+	if ((ieee80211_is_s1g_beacon(frame_control) &&
+	     !ies_mask->ies[WLAN_EID_S1G_BCN_COMPAT].ptr) ||
 	    (le16_to_cpu(frame_control) & IEEE80211_FC_COMPRESS_SSID)) {
 		ies_order_table = morse_ext_s1g_short_beacon_ies_order;
 		ies_order_table_len = ARRAY_SIZE(morse_ext_s1g_short_beacon_ies_order);
--- a/vendor_ie.c
+++ b/vendor_ie.c
@@ -139,8 +139,7 @@
  */
 static inline u8 *get_elements_from_s1g_beacon(struct ieee80211_ext *bcn)
 {
-	return (ieee80211_is_s1g_short_beacon(bcn->frame_control) ?
-		bcn->u.s1g_short_beacon.variable : bcn->u.s1g_beacon.variable);
+	return bcn->u.s1g_beacon.variable + ieee80211_s1g_optional_len(bcn->frame_control);
 }
 
 /**
